### DISCUSSION TITLE
Does Longhorn change ctime of the files in volume?

### DISCUSSION THREAD

**Discussion by jwenjian** [longhorn/longhorn#4130] [Q&A]
Hi,

I am facing an issue that `ctime` of the files in longhorn volume changes every time I start the pod.

`stat` command output:

```
  File: 'configuration.xml'
  Size: 16544           Blocks: 48         IO Block: 4096   regular file
Device: 810h/2064d      Inode: 12          Links: 1
Access: (0777/-rwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2022-06-17 15:29:56.000000000 +0800
Modify: 2022-06-15 17:08:49.000000000 +0800
Change: 2022-06-17 15:43:17.000000000 +0800
 Birth: -
```

Any hints?

Thanks.

**Comment by derekbit** [comment]
I tried it in my env, and the ctime was not modified.
Is it possible that your workload modifies the files' data or metadata?
Or, can you provide the steps to reproduce?

**Reply by jwenjian** [reply]
Thanks for your time, I will try to list more info below:

- env
  - one node cluster created by RKE2 on Redhat 8.4
  - SELinux enabled
  - Longhorn is installed by default, the pvc is attached to /dev/longhorn/pvc-x--x-x-x-x-x-x-x--xxx
 
My workload just read this file and check the file's ctime and mtime, no modification.

This issue not happened on the another env( longhorn is configured to attach data to a dedicated disk /data).

hope this helps and thanks in advance

**Reply by jwenjian** [reply]
I tried with host path volume, the `ctime` not changed anymore..

**Reply by jwenjian** [reply]
Additional info:

`kubectl version`:

```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.14+rke2r1", GitCommit:"57a3aa3f13699cf3db9c52d228c18db94fa81876", GitTreeState:"clean", BuildDate:"2021-12-15T21:39:40Z", GoVersion:"go1.15.15b5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.14+rke2r1", GitCommit:"57a3aa3f13699cf3db9c52d228c18db94fa81876", GitTreeState:"clean", BuildDate:"2021-12-15T21:39:40Z", GoVersion:"go1.15.15b5", Compiler:"gc", Platform:"linux/amd64"}
```

`longhorn version`:

```
longhornio/longhorn-manager:v1.2.3
```

`Host OS`:

```
# uname -a
Linux CNTHFRP176 4.18.0-305.el8.x86_64 #1 SMP Thu Apr 29 08:54:30 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by derekbit** [comment]
Probably related to the SELinux. Will the SELinux modify the extended attributes of the files in the volume? Can you check if it is caused by SELinux?

**Reply by jwenjian** [reply]
I disabled SELinux, the `ctime` still changed. :(

**Reply by derekbit** [reply]
Weird. How did you disable the SELinux?

**Reply by jwenjian** [reply]
Disable the SELinux by `setenforce 0`, then confirm with `getenforce`, showing **Permissive**

**Reply by derekbit** [reply]
Not really familiar with SELinux. The SELinux relabeling might still work under Permissive mode. Probably need to set it to be Disabled mode.

**Reply by derekbit** [reply]
https://documentation.suse.com/sles/12-SP4/html/SLES-all/cha-selinux.html

> enforcing=0
> This option puts SELinux in permissive mode. In this mode, SELinux is fully functional, but does not enforce any of the security settings in the policy. Use this mode for configuring your system. To switch on SELinux protection, when the system is fully operational, change the option to enforcing=1 and add SELINUX=enforcing in /etc/selinux/config.

**Reply by jwenjian** [reply]
Thanks, I'll try

**Reply by jwenjian** [reply]
Confirmed it's caused by SELinux, I disabled SELinux and reboot, then the `ctime` not changed any more.

But, what security label should I give to the longhorn data folder to avoid this issue? I mean, disable SELinux is not recommended especially for production.

**Reply by jwenjian** [reply]
Just for your information, the security label of the /dev/longhorn is in below:

```
# ls -lZ /dev/longhorn/
total 0
brw-rw----. 1 root root system_u:object_r:device_t:s0 8, 48 Jun 20 15:58 pvc-5ae8240f-f5fd-41e2-9e62-384f5a7eed36
brw-rw----. 1 root root system_u:object_r:device_t:s0 8, 16 Jun 20 15:58 pvc-6187afe5-c875-4941-8e46-6d1b4d1502c7
brw-rw----. 1 root root system_u:object_r:device_t:s0 8, 32 Jun 20 15:58 pvc-790b978c-6fe2-431c-99d1-2a6b6214e09b

```

If I want enable SELinux, to resolve this issue, how do I change the security label of this folder? any docs?

Thanks.

**Reply by derekbit** [reply]
Thank you for confirming. We need to investigate this further. Please follow up on the ticket.

**Reply by jwenjian** [reply]
Thanks for your help, do I need open an issue or just follow this discussion?

**Reply by derekbit** [reply]
Let me convert it to an issue.

**Reply by derekbit** [reply]
We will update on https://github.com/longhorn/longhorn/issues/4157. Thank you.
